packages/wrangler/src/metrics/commands.ts

```diff
@@ -0,0 +1,70 @@
+import chalk from "chalk";
+import { defineCommand, defineNamespace } from "../core";
+import { logger } from "../logger";
+import { getMetricsConfig, updateMetricsPermission } from "./metrics-config";
+
+defineNamespace({
+	command: "wrangler telemetry",
+	metadata: {
+		description: "📈 Configure whether Wrangler collects telemetry",
+		owner: "Workers: Authoring and Testing",
+		status: "stable",
+		hidden: true,
+	},
+});
+
+defineCommand({
+	command: "wrangler telemetry disable",
+	metadata: {
+		description: "Disable Wrangler telemetry collection",
+		owner: "Workers: Authoring and Testing",
+		status: "stable",
+	},
+	async handler() {
+		updateMetricsPermission(false);
+		logTelemetryStatus(false);
+		logger.log(
+			"Wrangler is no longer collecting telemetry about your usage.\n"
+		);
+		return;
+	},
+});
+
+defineCommand({
+	command: "wrangler telemetry enable",
+	metadata: {
+		description: "Enable Wrangler telemetry collection",
+		owner: "Workers: Authoring and Testing",
+		status: "stable",
+	},
+	async handler() {
+		updateMetricsPermission(true);
+		logTelemetryStatus(true);
+		logger.log(
+			"Wrangler is now collecting telemetry about your usage. Thank you for helping make Wrangler better 🧡\n"
+		);
+		return;
+	},
+});
+
+defineCommand({
+	command: "wrangler telemetry status",
+	metadata: {
+		description: "Check whether Wrangler telemetry collection is enabled",
+		owner: "Workers: Authoring and Testing",
+		status: "stable",
+	},
+	async handler(_, { config }) {
+		const metricsConfig = await getMetricsConfig({
+			sendMetrics: config.send_metrics,
+		});
```

Comment:
Would `readMetricsConfig` be enough? `getMetricsConfig` does more to get userId and will re-confirm whether users wanna enable telemetry in some cases which seems unnecessary here.

Reply:
oh yeah meant to flag this - should `status` be per-project? i.e. should it take into account send_metrics in wrangler.toml at all. If we strictly want to to be about user-level, we should probably clarify that in the messaging. And also we should probably clarify the order of precedence for wrangler.toml vs env variable vs set-by-command.

Also it does re-confirm currently, but I figure we would remove that when we switch to on-by-default.

Reply:
I would expect the status to show the actual stored preference—basically, what the telemetry command is modifying. But yeah, we should make sure the message is clear here.

Reply:
I feel like we should show two status: Global / Project (which reflects the send_metrics config?)

Reply:
what do you think about this - b44e1b5e3f7e67f54a826a502dc550c6a369991d 
And do you think we should explain how global/project is set underneath, or is just the status fine?

Reply:
Let's get some feedbacks from Nevi and Liz on this.

Reply:
resolved in 7bc5de1fa7707d6731d8e82d1b417f793f9fcc67 after discussions with Nevi/Liz